### PR TITLE
Add support for filtering change types when diffing

### PR DESCRIPTION
**Pull request by lukebjerring**
Progresses https://github.com/w3c/wptdashboard/issues/301

Uses a git-inspired filtering charset param.

?filter=[A][C][D]
A = Added
C = Changed
D = Deleted

### CODE REVIEW

File: run_diff.go [hunk 1]
```diff
@@ -124,32 +124,43 @@ func fetchRunResultsJSON(ctx context.Context, r *http.Request, run TestRun) (res
...
-			countDiff := abs(resultsBefore[1] - resultsAfter[1])
-			if countDiff == 0 && passDiff == 0 {
-				continue
-			}
-			// Changed tests is at most the number of different outcomes,
-			// but newly introduced tests should still be counted (e.g. 0/2 => 0/5)
-			diff[test] = []int{
-				max(passDiff, countDiff),
-				max(resultsBefore[1], resultsAfter[1]),
+	if filter.Deleted || filter.Changed {
```
Comment (lukebjerring) [inline run_diff.go]: This diff tool isn't great. FWIW, changes are just:
- Surround whole code-block in the if-statement above
- Add continue if-statements

File: params.go [hunk 2]
```diff
@@ -112,3 +112,43 @@ func ParseMaxCountParam(r *http.Request) (count int, err error) {
...
+	param = DiffFilterParam{
+		true,
+		true,
+		true,
+	}
+	if filter := r.URL.Query().Get("filter"); filter != "" {
+		param = DiffFilterParam{}
+		for _, char := range filter {
+			switch char {
+			case 'A':
```
Comment (foolip) [inline params.go]: Comment around here as well that this is inspired by `git --diff-filter=`, which, BTW, I really like :) I don't think that rename detection is in any way a straightforward thing, or likely to ever be worth doing, but if we do, it should be 'R'.

File: run_diff.go (hunk 1 above)
Comment (foolip) [reply run_diff.go]: Thanks, very helpful.

File: params.go (hunk 2 above)
Comment (lukebjerring) [reply params.go]: Noted, and Done.

**Pull request merged**